Feature update login page

ChaseVZ opened the pull request.

ChaseVZ committed:
imported context

ChaseVZ committed:
non-functional push for debugging purposes

ChaseVZ committed:
another debug push

ChaseVZ committed:
const classes line fixed

ChaseVZ committed:
cleaned up InputField component use

ChaseVZ committed:
fixed temp/user bug

ChaseVZ committed:
added basic error functionality. Conditional hardcoded for now

ChaseVZ committed:
basic dynamic error codes

ChaseVZ committed:
changed alerts to Dialogs

ChaseVZ committed:
cleaned up code

ChaseVZ committed:
small fixes

ChaseVZ committed:
Merge branch 'development' of https://github.com/d-mooers/The-Market-Client into feature-update-login-page

ChaseVZ committed:
text change

ChaseVZ committed:
add async function

d-mooers committed:
Fixed API for logging in

ChaseVZ committed:
simplified email-username to one field. Fixed GET call

ChaseVZ committed:
fixed a ton (seemingly all) bugs w/ resp code

ChaseVZ committed:
final code cleanup

d-mooers commented on client/src/components/LoginPage/InputField.js:
```diff
@@ -0,0 +1,39 @@
+
+const styles = makeStyles({
+  textBox: {
+    margin: 10,
```
Not required, but its generally better to use "rem" instead of pixels.  "rem" is alot more accessible as it will scale to the user's settings.  The default is 1rem = 16px, so for this it would be margin: "0.625rem"

ChaseVZ replied:
got it!

d-mooers commented on client/src/components/LoginPage/index.js:
```diff
@@ -1,86 +1,110 @@
+  const [emailError, setEmailError] = useState(false);
+  const [passwordError, setPasswordError] = useState(false);
+  const [error, setError] = useState(false);
+  const [temp, setTempUser] = useState({
```
Consider naming conventions here.  You have 'setTempUser' but the associated state variable is just "temp".  All of your other state variables / setters use correct conventions though so good job.

ChaseVZ replied:
yeah I was meaning to do so - good call

d-mooers commented on client/src/components/LoginPage/index.js:
```diff
@@ -1,86 +1,110 @@
-        password: user["password"],
-      });
+    if (name === "email") {
+      setTempUser({ ...temp, email: value });
```
Nice use of object destructuring

d-mooers commented on client/src/components/LoginPage/index.js:
```diff
@@ -1,86 +1,110 @@
 
-    if (resp === 0) {
+      // Query went through successfully
+      return resp;
```
rather than just returning resp, I would return:
{
     success: resp.status === 200,
     user: resp.data
}

and on error....
{
   success: false,
   err: e
}

d-mooers commented on client/src/components/LoginPage/index.js:
```diff
@@ -1,86 +1,110 @@
       props.history.push("/browse");
-    } else if (resp === -1) {
-      alert("Invalid login");
+      setUser(temp);
```
We need to set the user to be the data sent back from the login request.  That data will contain the auth-ID that allows us to authenticate future api calls. 
Currently this sets the user to be whatever was typed into the fields

d-mooers commented on client/src/components/LoginPage/index.js:
```diff
@@ -1,86 +1,110 @@
+  }
+
+  // when user submits to login
+  async function submitForm() {
```
consider adding a state variable for loading, that is set to true at the start of this function and then false at the end?

d-mooers commented on client/src/components/LoginPage/index.js:
```diff
@@ -1,86 +1,110 @@
+    // if both fields have something then check backend
+    // if resp.status === 200 then success
+    // if resp.status === 'undefined', then it's an error and return resp.response.status (401)
+    else {
```
I would consider adding this logic to your submitForm() function.  This could almost replace the second part of your submitForm function, if you did setUser(resp.data).

d-mooers commented on d:
```diff
@@ -0,0 +1,17 @@
+* [32mdevelopment[m
```
Whats this file?

ChaseVZ replied:
I have no idea

d-mooers reviewed (changes requested):
setUser needs to be called with the whole resp.data from the backend, as that is what contains the auth id.

ChaseVZ committed:
Updates to fulfill pull request changes

Dialog now conditionally renders components. Cleaned up code involving response from GET call. Added Loading Dialog.

d-mooers commented:
lgtm

d-mooers commented on client/src/components/LoginPage/index.js:
```diff
@@ -1,85 +1,103 @@
+      setPasswordError(true);
+      resp = -1;
+    }
+    if (resp === 0) resp = await getUser(tempUser.email, tempUser.password);
```
Where do we hash the password?

d-mooers reviewed (changes requested):
Need to hash the password before sending it to the backend.  basic Auth hash is not enough

slaterOwen reviewed (commented):
All updates looked good to me. I liked the comments you made in LoginPage index.js. Awesome job

ChaseVZ committed:
added sha256 password hashing

d-mooers reviewed (approved):
Looks great! Good job!

ChristianBloemhof20 commented on client/src/components/LoginPage/index.js:
```diff
@@ -1,85 +1,105 @@
-        password: user["password"],
-      });
+    if (name === "email") {
+      setTempUser({ ...tempUser, email: value });
```
I'm gonna need you to explain to me how this works because I was referencing your work when doing registration page and really liked how you did this but it simply would not work for me

ChaseVZ replied:
I'll DM you about it

ChristianBloemhof20 reviewed (commented):
Very clean page, good job!
